temporalio/bridge/src/client.rs

```diff
@@ -117,23 +121,43 @@ macro_rules! rpc_call {
 
 #[pymethods]
 impl ClientRef {
+    fn drop_client(&mut self) {
+        self.retry_client = None
+    }
```

Comment:
I'm not 100% sure if this is sufficient. Clients are `Clone` themselves, and share the underlying connection. It's entirely possible that on the Core side we might clone the client somewhere and then dropping the client ref won't actually seal the deal.

That said, there's also no readily available way to actually close the underlying connections Core side either. So, we might have to just attach a small caveat here in the docstrings or something.